askama_derive/src/generator.rs

```diff
@@ -802,9 +803,18 @@ impl<'a> Generator<'a> {
         // Since nodes must not outlive the Generator, we instantiate a nested `Generator` here to
         // handle the include's nodes. Unfortunately we can't easily share the `includes` cache.
 
-        let ctx = &self.contexts[path.as_path()];
-        let heritage = if !ctx.blocks.is_empty() || ctx.extends.is_some() {
-            Some(Heritage::new(ctx, self.contexts))
+        let child_ctx = &mut self.contexts[path.as_path()].clone();
```

Comment:
These changes should be squashed into the first commit. I review on a commit-by-commit basis, so PRs should have a clean commit history.

Please add some comments about what this code is trying to do/why it is necessary.

Reply:
Sure, use `git rebase --interactive main`, reorder the lines and replace the first word with `squash` (or `s`).

Reply:
done